src/app/providers/config/EmbeddedConfigContextProvider.tsx

```diff
@@ -1,14 +1,25 @@
-import React from "react";
-import { ConfigContext } from '@rhoas/app-services-ui-shared';
-import config from '../../../../config/config.json';
-import { EnvironmentConfigs, filterConfig } from "@app/providers/config/utils";
+import React from 'react';
+import { Config, ConfigContext } from '@rhoas/app-services-ui-shared';
+import configs from '../../../../config/config.json';
+import { EnvironmentConfigs, filterConfig } from '@app/providers/config/utils';
 
+declare const __webpack_public_path__: string;
 
 export const EmbeddedConfigProvider: React.FunctionComponent = ({ children }) => {
-  const c = config.config as EnvironmentConfigs;
-  return (
-    <ConfigContext.Provider value={filterConfig(c, config.federatedModules)}>
-      {children}
-    </ConfigContext.Provider>
-  );
-}
+  const [value, setValue] = React.useState<Config | undefined>(() => {
+    const [c, fetchConfig] = filterConfig(configs.config as EnvironmentConfigs, configs.federatedModules);
+    return fetchConfig ? undefined : c;
```

Comment:
If the environment doesn't force fetch the config the precompile config is loaded immideattly